funsor/einsum/numpy_log.py

```diff
@@ -0,0 +1,40 @@
+import numpy as np
```

Comment:
I believe [opt_einsum requires](https://optimized-einsum.readthedocs.io/en/stable/backends.html) also `tensordot` and `transpose` implementations, which you can get via
```py
from pyro.ops.einsum.util import Tensordot

tensordot = Tensordot(einsum)
transpose = np.transpose
```
These may not be exercised in existing tests, but I believe they will be required for more complex contractions in e.g. NumPyro's enumeration machinery.